packages/core/src/components/overlay/overlay.tsx

```diff
@@ -348,7 +349,19 @@ export class Overlay extends React.PureComponent<IOverlayProps, IOverlayState> {
     private handleDocumentClick = (e: MouseEvent) => {
         const { isOpen, onClose } = this.props;
         const eventTarget = e.target as HTMLElement;
-        const isClickInOverlay = this.containerElement != null && this.containerElement.contains(eventTarget);
+
+        const { openStack } = Overlay;
+        const stackIndex = openStack.indexOf(this);
+
+        const isClickInDescendantOverlay = openStack
+            .slice(stackIndex)
```

Comment:
i'm struggling to wrap my head around why this works... new ("descendant") Overlays are pushed to the end of the stack so we can slice off everything _before_ this one?

Reply:
yes